### PR TITLE
Address code smells within ingestion source

### PR DESCRIPTION
**Pull request by mithmatt**
### Describe your changes :

I worked on this to reduce the number of code smells reported by Sonar

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

 Ingestion: @harshach @ayush-shah @pmbrull

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/glue.py [hunk 1]
```diff
@@ -115,9 +115,9 @@ def next_record(self) -> Iterable[Entity]:
                 yield from self.ingest_tables()
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/source/glue.py:115]: Fixing variable naming convention within the file

File: ingestion/src/metadata/ingestion/source/sample_entity.py [hunk 2]
```diff
@@ -40,7 +40,6 @@
 from metadata.ingestion.ometa.ometa_api import OpenMetadata
 from metadata.ingestion.ometa.openmetadata_rest import MetadataServerConfig
 from metadata.ingestion.source.sql_source import SQLConnectionConfig
-from metadata.utils.helpers import snake_to_camel
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/source/sample_entity.py:43]: unused import

File: ingestion/src/metadata/ingestion/source/sample_entity.py [hunk 3]
```diff
@@ -40,7 +40,6 @@
 from metadata.ingestion.ometa.ometa_api import OpenMetadata
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/source/sample_entity.py:40]: Removing unused variables err and charts

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 4]
```diff
@@ -278,7 +278,6 @@ def fetch_tables(
                 except Exception as err:
                     logger.error(repr(err))
                     logger.error(err)
-                    pass
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/source/sql_source.py:281]: not required

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 5]
```diff
@@ -508,12 +507,13 @@ def _get_columns(
                     col_type = get_column_type(
                         self.status, dataset_name, column["type"]
                     )
-                    if col_type == "ARRAY":
-                        if re.match(r"(?:\w*)(?:\()(\w*)(?:.*)", str(column["type"])):
-                            arr_data_type = re.match(
-                                r"(?:\w*)(?:[(]*)(\w*)(?:.*)", str(column["type"])
-                            ).groups()
-                            data_type_display = column["type"]
+                    if col_type == "ARRAY" and re.match(
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/source/sql_source.py:510]: merge the if conditions

File: ingestion/src/metadata/ingestion/source/glue.py [hunk 6]
```diff
@@ -194,26 +194,28 @@ def ingest_tables(self, next_tables_token=None) -> Iterable[OMetaDatabaseAndTabl
...
-                    downstreamTasks.append(
-                        list(self.task_id_mapping.keys())[
-                            list(self.task_id_mapping.values()).index(
-                                edges["DestinationId"]
-                            )
-                        ][:128]
-                    )
-        return downstreamTasks
+            if (
+                edges["SourceId"] == task_unique_id
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/source/glue.py:200]: merged the if conditions with an and operation to reduce indentation

**Review by harshach — APPROVED** [review]

**Pull request merged**